packages/evm/contracts/test/arbitration/ArbitrableMock.sol

```diff
@@ -5,37 +5,25 @@ import "../../arbitration/IArbitrator.sol";
 
 
 contract ArbitrableMock is IArbitrable {
-    bytes4 public constant ERC165_INTERFACE = ERC165_INTERFACE_ID;
-    bytes4 public constant ARBITRABLE_INTERFACE = ARBITRABLE_INTERFACE_ID;
-
     IArbitrator internal arbitrator;
 
     constructor (IArbitrator _arbitrator) public {
         arbitrator = _arbitrator;
     }
 
-    function interfaceId() external pure returns (bytes4) {
-        IArbitrable iArbitrable;
-        return iArbitrable.submitEvidence.selector ^ iArbitrable.rule.selector;
-    }
-
     function createDispute(uint8 _possibleRulings, bytes calldata _metadata) external {
         (address recipient, ERC20 feeToken, uint256 disputeFees) = arbitrator.getDisputeFees();
         feeToken.approve(recipient, disputeFees);
         arbitrator.createDispute(_possibleRulings, _metadata);
     }
 
     function submitEvidence(uint256 _disputeId, bytes calldata _evidence, bool _finished) external {
-        emit EvidenceSubmitted(arbitrator, _disputeId, msg.sender, _evidence, _finished);
+        arbitrator.submitEvidence(_disputeId, msg.sender, _evidence);
         if (_finished) arbitrator.closeEvidencePeriod(_disputeId);
     }
 
-    function rule(uint256 _disputeId, uint256 _ruling) external {
-        emit Ruled(IArbitrator(msg.sender), _disputeId, _ruling);
-    }
-
-    function interfaceID() external pure returns (bytes4) {
-        IArbitrable arbitrable;
-        return arbitrable.submitEvidence.selector ^ arbitrable.rule.selector;
+    function rule(uint256 _disputeId) external {
+        (, uint256 ruling) = arbitrator.rule(_disputeId);
```

Comment:
If I understand correctly, the motivation behind this is to shift the control flow so a user would initiate rulings from the IArbitrable instance rather than going directly to the IArbitrator (e.g. "pull in the ruling" rather than "push out the ruling")?

And in terms of pending state, if we assume the IArbitrable never calls `arbitrator.rule()`, no other state in the IArbitrator is impacted? It's just forever "yeah, you can call me to fetch your ruling, but I don't care if you do or don't"?

Reply:
Exactly, this is basically removing the dependency of the Arbitrable in the Arbitrator, now is totally decoupled. The Arbitrable can call `rule` multiple times or not call it at all, the Arbitrator just doesn't care